hatice-ecevit-2016400138/assignment1/Anagram.java

```diff
@@ -0,0 +1,89 @@
+
+ import java.util.Arrays;
+ import java.util.HashMap;
+public class Anagram {
+
+    public static void main(String[] args) {
+        System.out.println(areAnagrams("apple","appll",true));
+        System.out.println(areAnagrams("listen","Silent",false));
+        System.out.println(areAnagramSentences("hi there hello","iH tHree lloeh",false));
+        System.out.println(areAnagramSentences("a a", "a b", false));
+    }
+
+    /**
+     * @param s1    First String
+     * @param s2    Second String
+     * @param caseSensitive true if the method should work in a case sensitive way, otherwise false
+     * @return  true if s1 and s2 are anagrams, false otherwise
+     */
+    public static boolean areAnagrams(String s1, String s2, boolean caseSensitive){
+        if (!caseSensitive) {
+            s1 = s1.toLowerCase();
+            s2 = s2.toLowerCase();
+        }
+        HashMap<Character,Integer> letterMap=new HashMap<Character, Integer>();
+        if (s1.length()!= s2.length())
+                return false;
+        else {
+            for (int i = 0;i<s1.length();i++) {
+                char currentChar = s1.charAt(i);
+                if (letterMap.containsKey(currentChar)) {
+                    int value = letterMap.get(currentChar);
+                    value++;
+                    letterMap.put(currentChar,value);
+                }
+                else {
+                    letterMap.put(currentChar,1);
+                }
+            }
+            for (int i = 0;i<s2.length();i++) {
+                char currentChar = s2.charAt(i);
+                if (!letterMap.containsKey(currentChar))
+                    return false;
+                else {
+                    int value = letterMap.get(currentChar);
+                    value--;
+                    letterMap.put(currentChar,value);
+                }
+            }
+            for (Integer value: letterMap.values()) {
+                if(value != 0)
+                    return false;
+            }
+            return true;
+        }
+    }
+
+    /**
+     * @param s1    First Sentence
+     * @param s2    Second Sentence
+     * @param caseSensitive true if the method should work in a case sensitive way, otherwise false
+     * @return  true if s1 and s2 are anagram sentences, false otherwise
+     */
+    public static boolean areAnagramSentences(String s1,String s2, boolean caseSensitive){
+        String[] words1 = s1.split("\\s+");
+        String[] words2 = s2.split("\\s+");
+        if (words1.length != words2.length) {
+            return false;
+        }
+        else {
+            boolean[] anagram1 = new boolean[words1.length];
+            boolean[] anagram2 = new boolean[words2.length];
+            Arrays.fill(anagram1,Boolean.FALSE);
+            Arrays.fill(anagram2,Boolean.FALSE);
+            for (int i = 0;i<words1.length;i++) {
+                for (int j = 0;j<words2.length;j++) {
```

Comment:
Even if every word in `words1` has an anagram in `words2` and vice-versa, is it possible for the two sentences not be anagrams of each other?

**Hint:**
Given the sentences `"a b b c"` and `"a b c c"`, what would your implementation return? Does it match what you would expect?